NIFI-6770 - Set validator to Validator.VALID to allow empty password …

thenatog opened the pull request:
…for truststores.

NIFI-6770 - Added no-password keystore for tests

Thank you for submitting a contribution to Apache NiFi.

Please provide a short description of the PR here:

#### Description of PR

_Enables X functionality; fixes bug NIFI-YYYY._

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically `master`)?

- [x] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [x] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [x] Have you written or updated unit tests to verify your changes?
- [x] Have you verified that the full build is successful on both JDK 8 and JDK 11?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [ ] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [ ] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

alopresto commented:
It looks like the code changes here only affect components using a `SSLContextService`. Did you evaluate if the framework already allows for passwordless truststores for `nifi.security.truststore` in _nifi.properties_?

thenatog commented:
Added another code change and unit test to allow for the NiFi framework to use passwordless truststores. @alopresto

alopresto commented:
Reviewing...

alopresto commented on nifi-nar-bundles/nifi-standard-services/nifi-ssl-context-bundle/nifi-ssl-context-service/src/main/java/org/apache/nifi/ssl/StandardSSLContextService.java:
```diff
@@ -70,7 +70,7 @@
             .defaultValue(null)
-            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .addValidator(Validator.VALID)
             .sensitive(true)
```
Added `.required(false)`.

alopresto commented:
While doing live PR review with @thenatog we discovered a number of legacy issues with the relevant code & unit tests. We have made additional changes available on this branch: https://github.com/alopresto/nifi/tree/NIFI-6770.

thenatog commented:
Pulled in the changes Andy made and ran a contrib check. Ran some smoke tests:

1. Tested insecure NiFi with an InvokeHTTP processor to https://nifi.apache.org/ with a passwordless truststore containing *.apache.org certificate.
    a. Removed certificate from passwordless truststore to see PKIX error "unable to find valid certification path to requested target"

1. Tested secure NiFi with an InvokeHTTP processor to https://nifi.apache.org/ with a passwordless truststore *.apache.org certificate. 
    a. Removed certificate from passwordless truststore to see PKIX error "unable to find valid certification path to requested target"

1. Tested secure NiFi with service-level passwordless truststore
